### PR TITLE
handle bundle upgrades

### PR DESCRIPTION
**Pull request by tchughesiv**
build example -
```bash
$ make generate bundle bundle-build catalog-build catalog-push \
    CATALOG_BASE_IMG=registry.developers.crunchydata.com/crunchydata/crunchy-bridge-operator-catalog:v0.0.2 \
    IMAGE_TAG_BASE=registry.developers.crunchydata.com/crunchydata/crunchy-bridge-operator \
    VERSION=0.0.4
```

Signed-off-by: Tommy Hughes <tohughes@redhat.com>

### CODE REVIEW

**Comment by jeremyary** [comment]
@wilybrace ready for review, thanks!

File: bundle/manifests/crunchy-bridge-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -432,4 +432,5 @@ spec:
   provider:
     name: Crunchy Data
     url: https://www.crunchydata.com
-  version: 0.0.2-dev
+  replaces: crunchy-bridge-operator.v0.0.2
```
Comment (wilybrace) [inline bundle/manifests/crunchy-bridge-operator.clusterserviceversion.yaml:435]: Are the values for this version-specific, or can it define replaces for a range? This change references v0.0.2, but v0.0.3 is the latest on our server.

File: config/manifests/bases/crunchy-bridge-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -60,4 +60,5 @@ spec:
   provider:
     name: Crunchy Data
     url: https://www.crunchydata.com
+  replaces: crunchy-bridge-operator.v0.0.2
```
Comment (wilybrace) [inline config/manifests/bases/crunchy-bridge-operator.clusterserviceversion.yaml:63]: Are the values for this version-specific, or can it define replaces for a range? This change references v0.0.2, but v0.0.3 is the latest on our server.

**Review by wilybrace — CHANGES_REQUESTED** [review]

File: bundle/manifests/crunchy-bridge-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (tchughesiv) [reply bundle/manifests/crunchy-bridge-operator.clusterserviceversion.yaml:435]: our issue is we didn't use 0.0.3 and that version doesn't have upgrade logic build into it, so clusters with 0.0.2 wouldn't upgrade to 0.0.4. that said, I'll look into implementing the `olm.skipRange` annotation for this... as it's the better approach should it work

File: config/manifests/bases/crunchy-bridge-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (tchughesiv) [reply config/manifests/bases/crunchy-bridge-operator.clusterserviceversion.yaml:63]: our issue is we didn't use 0.0.3 and that version doesn't have upgrade logic build into it, so clusters with 0.0.2 wouldn't upgrade to 0.0.4. that said, I'll look into implementing the `olm.skipRange` annotation for this... as it's the better approach should it work

**Comment by wilybrace** [comment]
If v0.0.3 wasn't used, I don't see an issue identifying v0.0.2 as the replacement and treating v0.0.3 as a "dead" version.

**Pull request merged**